config/environment.js

```diff
@@ -1,71 +1,201 @@
-var url = require('url');
+const url = require("url");
 
-function isTrue(value) {
-  value = String(value).toLowerCase();
-  return value === 'true';
+const possibleENVS = [
+  "IRENE_API_HOST",
+  "IRENE_DEVICEFARM_URL",
+  "IRENE_DEVICEFARM_HOST",
+  "IRENE_API_SOCKET_PATH",
+  "IRENE_ENABLE_REGISTRATION",
+  "IRENE_REGISTRATION_LINK",
+  "IRENE_SHOW_LICENSE",
+  "IRENE_ENABLE_CRISP",
+  "IRENE_ENABLE_HOTJAR",
+  "IRENE_ENABLE_PENDO",
+  "IRENE_ENABLE_CSB",
+  "IRENE_ENABLE_MARKETPLACE",
+  "IRENE_ENABLE_ROLLBAR",
+  "ENTERPRISE",
+  "WHITELABEL_ENABLED",
+  "WHITELABEL_NAME",
+  "WHITELABEL_LOGO",
+  "WHITELABEL_THEME",
+];
+
+const ENVHandlerCONST = {
+  possibleENVS: possibleENVS,
+  defaults: {
+    IRENE_API_HOST: "https://api.appknox.com",
+    IRENE_DEVICEFARM_URL: "wss://devicefarm.appknox.com",
+    IRENE_DEVICEFARM_HOST: "https://devicefarm.appknox.com",
+    IRENE_API_SOCKET_PATH: "https://socket.appknox.com",
+    IRENE_ENABLE_REGISTRATION: false,
+    IRENE_REGISTRATION_LINK: "",
+    IRENE_SHOW_LICENSE: false,
+    IRENE_ENABLE_CRISP: false,
+    IRENE_ENABLE_HOTJAR: false,
+    IRENE_ENABLE_PENDO: false,
+    IRENE_ENABLE_CSB: false,
+    IRENE_ENABLE_MARKETPLACE: false,
+    IRENE_ENABLE_ROLLBAR: false,
+    ENTERPRISE: false,
+    WHITELABEL_ENABLED: false,
+    WHITELABEL_NAME: "",
+    WHITELABEL_LOGO: "",
+    WHITELABEL_THEME: "dark",
+  },
+  pluginENVS: [
+    "IRENE_ENABLE_CRISP",
+    "IRENE_ENABLE_HOTJAR",
+    "IRENE_ENABLE_PENDO",
+    "IRENE_ENABLE_CSB",
+    "IRENE_ENABLE_MARKETPLACE",
+    "IRENE_ENABLE_ROLLBAR",
+  ],
+  processENV: Object.keys(process.env).reduce((acc, key) => {
+    if (possibleENVS.indexOf(key) > -1) {
+      acc[key] = process.env[key];
+    }
+    return acc;
+  }, {}),
+};
+
+class ENVHandler {
```

Comment:
both initializer and env has the same ENVHandler class implementation, is that required to be separated?

Reply:
nope, i have no idea how to unify them. I need ENVHandler in both environment.js and in initializer, but didn't find a way to share it from a single place.